df.knowledge.xml

```diff
@@ -382,11 +382,396 @@
         <flag-bit name='engineering_machine_trip_hammer'/>
     </bitfield-type>
 
+    <struct-type type-name='knowledge_scholar_category_flag' custom-methods='true'>
+        <int32_t name='category' comment='determines which bitflags to use'/>
+        <uint32_t name='flags' comment='one of the 14 bitflag structs'/>
+        <!-- the compound below does not work because bitfields have constructors -->
+<!--
+        <compound name='flags' is-union='true'>
+            <compound name='flags_0' type-name='knowledge_scholar_flags_0'/>
+            <compound name='flags_1' type-name='knowledge_scholar_flags_1'/>
+            <compound name='flags_2' type-name='knowledge_scholar_flags_2'/>
+            <compound name='flags_3' type-name='knowledge_scholar_flags_3'/>
+            <compound name='flags_4' type-name='knowledge_scholar_flags_4'/>
+            <compound name='flags_5' type-name='knowledge_scholar_flags_5'/>
+            <compound name='flags_6' type-name='knowledge_scholar_flags_6'/>
+            <compound name='flags_7' type-name='knowledge_scholar_flags_7'/>
+            <compound name='flags_8' type-name='knowledge_scholar_flags_8'/>
+            <compound name='flags_9' type-name='knowledge_scholar_flags_9'/>
+            <compound name='flags_10' type-name='knowledge_scholar_flags_10'/>
+            <compound name='flags_11' type-name='knowledge_scholar_flags_11'/>
+            <compound name='flags_12' type-name='knowledge_scholar_flags_12'/>
+            <compound name='flags_13' type-name='knowledge_scholar_flags_13'/>
+        </compound>
+-->
+
+        <!-- so that the header generator includes the header we need. -->
+        <static-array name='dummy' count='0'>
```

Comment:
I'm not sure if count=0 will compile everywhere (getting Travis to check an updated submodule is tricky, yeah).
Can't you just include dfhack_knowledge_scholar_flag.h in knowledge_scholar_category_flag.methods.inc?